src/libs/ReportUtils.ts

```diff
@@ -5473,6 +5473,40 @@ function temporary_getMoneyRequestOptions(
     return getMoneyRequestOptions(report, policy, reportParticipants, canUseTrackExpense, true) as Array<Exclude<IOUType, typeof CONST.IOU.TYPE.REQUEST | typeof CONST.IOU.TYPE.SEND>>;
 }
 
+/**
+ * Invoice sender, invoice receiver and auto-invited admins cannot leave
+ */
+function canJoinOrLeaveInvoiceRoom(report: OnyxEntry<Report>): boolean {
```

Comment:
I like that this has been moved. But confused about why we need to consider it for the "join" case.

Can we actually _join_ an invoice room? What state are things in when we do that? Maybe an example would help.

Reply:
P.S. if we are doing this you have a lot of duplicate code happening below that needs to be moved.

Reply:
> I like that this has been moved. But confused about why we need to consider it for the "join" case.

Yes you're right, the case of invoice room is used in `ReportDetailPage` and we can only leave this room.

Reply:
@marcaaron I checked again and you're right, we don't have the "join" case for invoice room. Updated the function.